devito/types/dense.py

```diff
@@ -958,15 +958,17 @@ def __init_finalize__(self, *args, **kwargs):
         else:
             raise TypeError("`space_order` must be int or 3-tuple of ints")
 
-        # Dynamically add derivative short-cuts
-        self._fd = generate_fd_shortcuts(self)
-
+        self.__fd_setup__()
```

Comment:
no, it should be:

```
self._fd = self.__fd_setup__()
```

that is, the assignment should be explicitly visible in `__init__`, not buried in the private method